src/3rd_party/pathie-cpp/src/entry_iterator.cpp

```diff
@@ -31,7 +31,7 @@
 #include "../include/path.hpp"
 #include "../include/errors.hpp"
 
-#if defined(__unix__)
+#if defined(__unix__) || defined(__APPLE__)
```

Comment:
I do not like this define __APPLE__ too much (if we adopt this pattern, it will proliferate if we add more platforms). Is there a common symbol? Why is __unix__ not defined on Mac?

Reply:
I have no idea. If you have a mac lying around, you can try to do `gcc -dM -E - </dev/null`
Neither clang, nor gcc define `unix` on mac. I am open to suggestions. Looking through fbgemm code, they use `__linux__` and `__APPLE__` defines.

Reply:
Can we use `BSD` instead of `__APPLE__`?

Reply:
In this particular case, we can. Pathie was designed to not compile unless the operating system was supported. There are multiple (non consistent!) defines throughtout the codebase that check for windows and have an else case `#error Unsupported system.` that doesn't compile. I can change this code to:
```c++
#ifdef _WIN32
#include <Windows.h>
#else
#include <sys/types.h>
#include <dirent.h>
#include <errno.h>
#include <stdexcept>
#endif
```
But this further deviates from upstream.